lib/pages/home/feedback_nudge.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:acs_upb_mobile/authentication/service/auth_provider.dart';
+import 'package:acs_upb_mobile/pages/class_feedback/service/feedback_provider.dart';
+import 'package:acs_upb_mobile/pages/class_feedback/view/classes_feedback_checklist.dart';
+import 'package:acs_upb_mobile/pages/classes/model/class.dart';
+import 'package:acs_upb_mobile/pages/classes/service/class_provider.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class FeedbackNudge extends StatefulWidget {
+  @override
+  _FeedbackNudgeState createState() => _FeedbackNudgeState();
+}
+
+class _FeedbackNudgeState extends State<FeedbackNudge> {
+  Set<ClassHeader> userClasses;
+  Map<String, dynamic> userClassesFeedbackProvided;
+  String feedbackFormsLeft;
+
+  // TODO(AndreiMirciu): Find a better approach on how to calculate the number of feedback forms that need to be completed
+  Future<void> getUserClasses() async {
+    final ClassProvider classProvider =
+        Provider.of<ClassProvider>(context, listen: false);
+    final authProvider = Provider.of<AuthProvider>(context, listen: false);
+    final feedbackProvider =
+        Provider.of<FeedbackProvider>(context, listen: false);
+
+    await classProvider.fetchClassHeaders(uid: authProvider.uid).then((value) {
+      if (mounted) setState(() => userClasses = value.toSet());
+    });
+    await feedbackProvider
```

Comment:
There's really no point in calling then with await in an async function. Just do value = await function() and then use the value, here and below.